include/boost/url/impl/segments.hpp

```diff
@@ -37,25 +37,23 @@ class segments::iterator
 
     iterator(
         string_view s,
-        std::size_t nseg,
-        const_string::factory const& a) noexcept
-        : impl_(s, nseg, a)
+        std::size_t nseg) noexcept
+        : impl_(s, nseg)
     {
     }
 
     // end ctor
     iterator(
         string_view s,
         std::size_t nseg,
-        const_string::factory const& a,
         int) noexcept
-        : impl_(s, nseg, a, 0)
+        : impl_(s, nseg, 0)
     {
     }
 
 public:
-    using value_type = const_string;
-    using reference = const_string;
+    using value_type = pct_encoded_view;
```

Comment:
Well no you can't have a reference as the `value_type`. This expression must be valie:

```
segments segs = ...,;

segments::value_type s( *segs.begin() );
```

At this point `s` should have ownership of a copy of the decoded segment. That means that `value_type` has to be `std::string`, or something like it.

Reply:
> std::string, or something like it.

We don't have `const_string` anymore, so `std::string`? 

Are we returning with the allocator parameter for `url_view::segments` then?

> Well no you can't have a reference as the value_type

That's not necessarily a usual requirement of views / or views of views.

Reply:
> We don't have const_string anymore, so std::string?

Yes

> Are we returning with the allocator parameter for url_view::segments then?

No

Reply:
OK. I changed that. But

```cpp
segments::value_type s( *segs.begin() );
```

still won't work. 

We would need

```cpp
segments::value_type s( (*segs.begin()).to_string() );
```

unless we return with the `explicit` `operator basic_string` you told me to remove.

Reply:
Yep... we might have to put that back unfortunately.